interface/operations/test/operations_client_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+// Common code for operations client data driven test suite
+package operations_test
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/parallaxsecond/parsec-client-go/interface/requests"
+)
+
+// Constants used in test descriptions
+const succeedString = "succeed"
+const failString = "fail"
+
+// Load test suite from json file
+func loadTestSuite(filename string) *TestSuite {
+	jsonfile, err := os.Open("data/" + filename)
+	if err != nil {
+		panic(err)
+	}
+	byteValue, err := ioutil.ReadAll(jsonfile)
+	if err != nil {
+		panic(err)
+	}
+
+	var testSuite TestSuite
+	err = json.Unmarshal(byteValue, &testSuite)
+	if err != nil {
+		panic(err)
+	}
+	return &testSuite
+}
+
+func unmarshalJSONObject(from, to interface{}) error {
+	jsonData, err := json.Marshal(from)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(jsonData, to)
+}
+
+type TestCase struct {
+	Name                  string      `json:"name"`
+	RequestData           interface{} `json:"request_data"`
+	ExpectedRequestBinary string      `json:"expected_request_binary"`
+	ResponseBinary        string      `json:"response_binary"`
+	ExpectedResponseData  interface{} `json:"expected_response"`
+	ExpectSuccess         bool        `json:"expect_success"`
+}
```

Comment:
Just for my understanding, with those tests you check that the stack correctly performs the conversions:
* from high-level functions with specific parameters (in `RequestData`) to binary data transmitted
* from the binary response to the high-level functions return value

Reply:
yes, that's correct, but a comment would be a good idea ;)